proxy/src/main/java/net/md_5/bungee/BungeeBossBar.java

```diff
@@ -0,0 +1,285 @@
+    public boolean addPlayer(ProxiedPlayer player)
+    {
+        Preconditions.checkNotNull( player, "player" );
+        if ( !players.contains( player ) )
+        {
+            players.add( player );
+        }
+        if ( player.isConnected() && visible )
+        {
+            return sendPacket( player, addPacket() );
+        }
+        return false;
+    }
+
+    @Override
+    public void addPlayers(Iterable<ProxiedPlayer> players)
+    {
+        Preconditions.checkNotNull( players, "players" );
+        for ( ProxiedPlayer player : players )
+        {
+            addPlayer( player );
+        }
+    }
+
+    @Override
+    public boolean removePlayer(ProxiedPlayer player)
+    {
+        Preconditions.checkNotNull( player, "player" );
+        if ( !players.contains( player ) )
+        {
+            return false;
+        }
+        players.remove( player );
+        if ( player.isConnected() && visible )
+        {
+            return sendPacket( player, removePacket() );
+        }
+        return false;
+    }
+
+    @Override
+    public void removePlayers(Iterable<ProxiedPlayer> players)
+    {
+        Preconditions.checkNotNull( players, "players" );
+        for ( ProxiedPlayer player : players )
+        {
+            removePlayer( player );
+        }
+    }
+
+    @Override
+    public void removeAllPlayers()
+    {
+        removePlayers( ImmutableList.copyOf( players ) );
+        players.clear();
+    }
+
+    @Override
+    public Collection<ProxiedPlayer> getPlayers()
+    {
+        return ImmutableList.copyOf( players );
+    }
+
+    @Override
+    public void setTitle(BaseComponent[] title)
+    {
+        this.title = Preconditions.checkNotNull( title, "title" );
+        if ( visible )
+        {
+            BossBar packet = new BossBar( uuid, 3 );
+            packet.setTitle( ComponentSerializer.toString( title ) );
+            sendToAffected( packet );
+        }
+    }
+
+    @Override
+    public void setHealth(float health)
+    {
+        Preconditions.checkArgument( 0 <= health && health <= 1, "Health may not be lower than 0 or greater than 1" );
+        this.health = health;
+        if ( visible )
+        {
+            BossBar packet = new BossBar( uuid, 2 );
+            packet.setHealth( health );
+            sendToAffected( packet );
+        }
+    }
+
+    @Override
+    public void setColor(BossBarColor color)
+    {
+        this.color = Preconditions.checkNotNull( color, "color" );
+        if ( visible )
+        {
+            setDivisions( color, division );
+        }
+    }
+
+    @Override
+    public void setDivision(BossBarDivision division)
+    {
+        this.division = Preconditions.checkNotNull( division, "division" );
+        if ( visible )
+        {
+            setDivisions( color, division );
+        }
+    }
+
+    @Override
+    public void setVisible(boolean visible)
+    {
+        boolean previous = this.visible;
+        if ( previous && !visible )
```

Comment:
Can be shortened to ```if previous != visible```

Reply:
It is ok in its current state.

Reply:
With your suggestion it'd need one more if anyway because create is not remove

Reply:
```
if ( visible != this.visible )
{
     sendPacket( visible ? createPacket : destroyPacket )
{
```